x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/create_field/create_field.tsx

```diff
@@ -123,6 +140,40 @@ export const CreateField = React.memo(function CreateFieldComponent({
     );
   }, [form, fieldTypeValue]);
 
+  const handleSemanticText = (data: FieldWithSemanticTextInfo) => {
+    if (data.inferenceId === undefined) {
+      return;
+    }
+
+    const inferenceData = inferenceToModelIdMap?.[data.inferenceId];
+
+    if (!inferenceData) {
+      return;
+    }
+
+    const { trainedModelId, defaultInferenceEndpoint, isDeployed } = inferenceData;
+
+    if (trainedModelId && defaultInferenceEndpoint) {
+      api?.createTextEmbeddingInference(data.inferenceId, trainedModelId);
+    }
+
+    if (!isDeployed && trainedModelId) {
+      ml?.mlApi?.trainedModels.startModelAllocation(trainedModelId);
+      const toast = {
```

Comment:
Do we want to call `ml` plugins api's from here directly or may be worth writing a new one for index management plugin?

cc: @sphilipse

Reply:
Using the ml functions is generally better.